--- Makefile.orig	2024-05-23 13:05:32.608449807 +0800
+++ Makefile	2024-05-23 13:06:06.735766002 +0800
@@ -554,7 +554,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2 -pthread -pthread -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls
+CFLAGS = -g -O2 -pthread -pthread -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -fno-reorder-blocks-and-partition
 CPP = gcc -E
 CPPFLAGS = 
 CSCOPE = cscope
@@ -577,7 +577,7 @@
 INSTALL_PROGRAM = ${INSTALL}
 INSTALL_SCRIPT = ${INSTALL}
 INSTALL_STRIP_PROGRAM = $(install_sh) -c -s
-LDFLAGS = 
+LDFLAGS = -Wl,--emit-relocs
 LIBOBJS = 
 LIBS = -levent 
 LTLIBOBJS = 
